text/0155-external-modules.md

```diff
@@ -0,0 +1,132 @@
+# Ecosystem Management
+
+Proposal how to leverage wider community contributing instrumentations and other packages.
+
+## Motivation
+
+For OpenTelemetry to become a de-facto standard in observability there must exist a vast ecosystem of OpenTelemetry components,
+including integrations with various libraries and frameworks in all languages supported by OpenTelemetry.
+We cannot possibly expect that all these integrations will be provided by the core maintainers of OpenTelemetry.
+We hope that wider community will integrate their projects with OpenTelemetry.
+We have to encourage that by providing great documentation, examples and tooling to integration authors,
+while still providing our end-users with some way to discover all available OpenTelemetry components together with some visibility into their quality.
+
+## Explanation
+
+The [OpenTelemetry Registry](https://opentelemetry.io/registry/) serves as a central catalogue of all known OpenTelemetry components,
+both provided by core maintainers of the project and any third party.
+
+In order for a component to be included into Registry its authors have to fill a [self-assessment form](#registry-self-assessment-form).
+
+The registry should allow a clear visibility of component's ownership, quality and compatibility with other OpenTelemetry components.
+
+A component can be removed from the Registry if any declaration from the self-assessment form is violated and not remedied
+in a timely manner, provisionally one or two weeks.
```

Comment:
1-2 weeks is actually not a lot of time. A lot of open-source work is not done as part of someone's day job, and I worry that this would be a high-pressure tactic on folks who are contributing in their spare time, for no compensation. With a deadline of 1-2 weeks, it feels plausible that people may not even see the notification for a few days and thus significantly shorten their window to respond, not even counting the time it would take to actually remedy any deficiencies.

I would suggest 4 weeks as a starting point (maybe a little more, maybe a little less). That should give folks plenty of time to see the notice (and clarify anything that needs), and then time to actually correct any problems.

Reply:
Quite a valid point. I'd be up for 3 weeks instead. @iNikem any opinion on this?

Reply:
For me, I think 3 weeks is much more reasonable. And we can always revisit the length of time later based on maintainer feedback. 😄

Reply:
all of this can be decided when this proposal becomes part of the specification, correct?

Reply:
These 1-2 weeks here are "provisional", meaning they are meant to indicate a rough estimate. E.g. "not 1 day and not 2 months, somewhere in between"

Reply:
I think it's also fine to resolve if/when merged into the specification; however it may be useful to our future selves to have this information in the OTEP now rather than later. Specifically if `1-2 weeks` means `not 1 day and not 2 months, somewhere in between` then that might be something other readers cannot know without finding this specific comment thread.

Reply:
Conversion of OTEP into specification was meant to be more of a mechanical process, not another round of debates. I would recommend resolving this issue now.

I agree 2 weeks is too short. It actually does not need to be that short for the specific goals, because a warning can be added to the registry much faster to alert people of an issue with some contrib module, and even override the specific item in self-assessment.

Reply:
Increased to "3 to 4"